src/main/java/com/o19s/es/ltr/feature/store/ScriptFeature.java

```diff
@@ -0,0 +1,193 @@
+                context.getQueryShardContext().lookup())), supplierSupplier);
+    }
+
+    static class LtrScript extends Query {
+        private final ScriptScoreFunction scoreFunction;
+        private final SupplierSupplier supplierSupplier;
+
+        LtrScript(ScriptScoreFunction scoreFunction, SupplierSupplier supplierSupplier) {
+            this.scoreFunction = scoreFunction;
+            this.supplierSupplier = supplierSupplier;
+        }
+
+        @Override
+        public String toString(String field) {
+            return "TODO";
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            // TODO:
+            return false;
+        }
+
+        @Override
+        public int hashCode() {
+            //TODO:
+            return 0;
+        }
+
+        @Override
+        public Weight createWeight(IndexSearcher searcher, boolean needsScores, float boost) throws IOException {
+            return new LtrScriptWeight(this);
+        }
+
+        class LtrScriptWeight extends FeatureVectorWeight {
+
+            protected LtrScriptWeight(Query query) {
+                super(query);
+            }
+
+            @Override
+            public Explanation explain(LeafReaderContext context, LtrRanker.FeatureVector vector, int doc) throws IOException {
+                LtrScript.this.supplierSupplier.set(() -> vector);
+                // TODO; figure out why we need to pass an explanation here
+                return scoreFunction.getLeafScoreFunction(context).explainScore(doc, Explanation.match(1F, "TODO"));
+            }
+
+            @Override
+            public Scorer scorer(LeafReaderContext context, Supplier<LtrRanker.FeatureVector> vectorSupplier) throws IOException {
+                LtrScript.this.supplierSupplier.set(vectorSupplier);
+                LeafScoreFunction leafScoreFunction = scoreFunction.getLeafScoreFunction(context);
+                DocIdSetIterator iterator = DocIdSetIterator.all(context.reader().maxDoc());
+                return new Scorer(this) {
+                    @Override
+                    public int docID() {
+                        return iterator.docID();
+                    }
+
+                    @Override
+                    public float score() throws IOException {
+                        return (float) leafScoreFunction.score(iterator.docID(), 0F);
+                    }
+
+                    @Override
+                    public DocIdSetIterator iterator() {
+                        return iterator;
+                    }
+                };
+            }
+
+            @Override
+            public void extractTerms(Set<Term> terms) {
+            }
+
+            @Override
+            public boolean isCacheable(LeafReaderContext ctx) {
+                return false;
+            }
+        }
+    }
+
+
+    static class SupplierSupplier implements Supplier<LtrRanker.FeatureVector> {
```

Comment:
I think we can fix the runtime dependency by implementing `Map<String, Float>` here.
A bunch of methods will return throw new UnsupportedOperationException but `get(String featureName)` and `containsKey(String featureName)` can be implemented and should be all what we need.

Reply:
@nomoa implementing `Map` seems to work. If you think this approach makes sense I can work on adding some tests and maybe more cleanup of code.

Reply:
@umeshdangat great!
yes I think the code deserves some cleanup (esp. better naming: SupplierSupplier is really not the most obvious class name :).
For the tests I think it'd be interesting to test : 
- a custom java script plugin (You can inject your own test plugin)
- possibly a painless script feature that uses a couple features exposed through this map

as for usability we have the same problem that we have for derived expressions, users have to declare these kind of features after the dependent features, e.g. if the script is declared at ordinal 1 but the feature it depends on is declared at ordinal 3 it won't work. I have no clue how to make this more obvious for users.

Thanks!